middleware/shell.go

```diff
@@ -0,0 +1,69 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+type Shell struct {
+	mu           sync.Mutex
+	Cmd          string
+	Args         []string
+	IntervalSecs int64
+	TimeoutSecs  int64
```

Comment:
could you make these `time.Duration`s all the way through so that it's clear from the calling code what we're setting (rather than potential confusion over the unit)

Reply:
Yeah good suggestion, will do.